### PR TITLE
Bug: Fix Follow and Unfollow Functionality

### PR DESCRIPTION
**Pull request by jsvelte**
### **Asana Link** 

- https://app.asana.com/0/1202627219907022/1202818098438824/f

### **Definition of Done**

- [x] [FE] Filtered user array if it is existing during the loop and return 'Follow' or 'Unfollow' string type

### **Notes**
- Currently, `breeze-next` recommend using localhost during local development of backend and frontend to avoid CORS "Same-Origin" issues.
Ref: https://github.com/laravel/breeze-next

### **Scenarios/Test Cases**

cd backend
- php artisan serve

cd frontend
- yarn || npm install
- yarn dev || npm run dev

### **Screenshots**
image

### CODE REVIEW

File: frontend/src/helpers/follow.ts [hunk 1]
```diff
@@ -20,17 +20,19 @@ export const useFollow = () => {
 
   const followStatus = (props: FollowStatusProps): string => {
     const { user, author } = props
-    const current_id = user?.followers?.map(({ id }) => id)
-    return current_id == author?.id ? 'Unfollow' : 'Follow'
+    const current_id: any = user?.followers?.map(({ id }) => id)
+    const filtered_user = current_id.filter((user_id: number) => user_id === author?.id) 
+    return filtered_user.length === 0 ? 'Follow' : 'Unfollow'
```
Comment (ziontavera) [inline frontend/src/helpers/follow.ts]:
```suggestion
    return !filtered_user.length ? 'Follow' : 'Unfollow'
```

File: frontend/src/helpers/follow.ts [hunk 2]
```diff
@@ -20,17 +20,19 @@ export const useFollow = () => {
...
   const handleFollow = async (props: HandleFollowProps): Promise<void> => {
     const { user, author, mutate, setLoading } = props
   
     setLoading(true)
-    const current_id = user?.followers?.map(({ id }) => id)
+    const current_id: any = user?.followers?.map(({ id }) => id)
+    const filtered_user = current_id.filter((user_id: number) => user_id === author?.id) 
   
-    if (current_id == author?.id) {
+    if (filtered_user.length !== 0) {
```
Comment (ziontavera) [inline frontend/src/helpers/follow.ts]:
```suggestion
    if (!!filtered_user.length) {
```

File: frontend/src/pages/dashboard.tsx [hunk 3]
```diff
@@ -102,7 +102,11 @@ const Dashboard: NextPage = (): JSX.Element => {
                       <Spinner className="w-6 h-6 text-orange-500" />
                     </div>
                   ) : (
-                    <DashboardList activities={user?.activities} /> 
+                    <>
+                      {user?.activities.length === 0 ? (
```
Comment (ziontavera) [inline frontend/src/pages/dashboard.tsx]:
```suggestion
                      {!user?.activities.length ? (
```

File: frontend/src/helpers/follow.ts (hunk 2 above)
Comment (jsvelte) [reply frontend/src/helpers/follow.ts]: Niccee <3

**Review by ziontavera — APPROVED** [review]

**Pull request merged**